alignak/daemons/brokerdaemon.py

```diff
@@ -207,87 +208,71 @@ def get_links_from_type(self, d_type):
+            return
+        if s_id not in links:
+            logger.warning("Unknown identifier '%s' for the %s connection!", s_id, s_type)
             return
 
-        # default timeout for daemons like pollers/reactionners/...
-        timeout = 3
-        data_timeout = 120
+        link = links[s_id]
+        logger.debug("- found: %s", link)
 
-        if i_type == 'scheduler':
+        if s_type == 'scheduler':
             # If sched is not active, I do not try to init
             # it is just useless
-            is_active = links[s_id]['active']
+            is_active = link['active']
             if not is_active:
+                logger.warning('Scheduler is not active, '
+                               'do not initalize its connection! Link: %s', link)
                 return
-            # schedulers also got real timeout to respect
-            timeout = links[s_id]['timeout']
-            data_timeout = links[s_id]['data_timeout']
 
         # If we try to connect too much, we slow down our tests
-        if self.is_connection_try_too_close(links[s_id]):
+        if self.is_connection_try_too_close(link, delay=5):
+            logger.debug("Too close connection retry, postponed")
             return
 
-        # Ok, we can now update it
-        links[s_id]['last_connection'] = time.time()
+        logger.info("Initializing connection with %s (%s)", link['name'], s_id)
+
+        # Get timeout for the daemon link (default defined in the satellite link...)
+        timeout = link['timeout']
+        data_timeout = link['data_timeout']
+
+        # Ok, we now update our last connection attempt
+        # and we increment the number of connection attempts
+        link['connection_attempt'] += 1
+        link['last_connection'] = time.time()
 
-        running_id = links[s_id]['running_id']
-        uri = links[s_id]['uri']
+        running_id = link['running_id']
+
+        # Create the HTTP client for the connection
         try:
-            con = links[s_id]['con'] = HTTPClient(uri=uri,
-                                                  strong_ssl=links[s_id]['hard_ssl_name_check'],
-                                                  timeout=timeout, data_timeout=data_timeout)
-        except HTTPEXCEPTIONS, exp:  # pragma: no cover, simple protection
-            # But the multiprocessing module is not compatible with it!
-            # so we must disable it immediately after
-            logger.warning("Connection problem to the %s %s: %s",
-                           i_type, links[s_id]['name'], str(exp))
-            links[s_id]['con'] = None
+            con = link['con'] = HTTPClient(uri=link['uri'],
+                                           strong_ssl=link['hard_ssl_name_check'],
+                                           timeout=timeout, data_timeout=data_timeout)
+        except HTTPClientConnectionException as exp:  # pragma: no cover, simple protection
```

Comment:
Hmmm... not sure about this. I am looking... breaking the connection when a bad connetion or time out happens is not necessarily a good idea!